eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/EnforceFinalFieldsCheck.java

```diff
@@ -0,0 +1,248 @@
+                return variableNameToken;
+            }
+        }
+
+        return null;
+    }
+
+    /*
+    * Saves a field name to a container depending on the provided type
+    */
+    private void saveField(final String fieldName, final int scopeParentType) {
+        if (scopeParentType == TokenTypes.METHOD_DEF) {
+            assignmentsFromMethods.add(fieldName);
+        } else if (scopeParentType == TokenTypes.CTOR_DEF) {
+            assignmentsFromConstructor.add(fieldName);
+        }
+    }
+
+    /*
+     * Review an assignation to save fields that gets assigned in constructor or in any method
+     *
+     * @param assignationToken an assignation token
+     */
+    private void checkAssignation(final DetailAST assignationToken) {
+        if (scopeParent == null || assignationToken.getChildCount() == 0) {
+            // not inside any method or constructor definition. No need to check anything
+            // or this is an assignation from a notation like @Test(timeout = 5000) where assignation has not ChildCount
+            return;
+        }
+
+        final DetailAST assignationParent = assignationToken.getParent();
+        if (assignationParent != null && TokenTypes.VARIABLE_DEF == assignationParent.getType()) {
+            // Assignation for a variable definition. No need to check this assignation
+            return;
+        }
+
+        DetailAST fieldToken = getAssignedField(assignationToken);
+
+        if (fieldToken != null) {
+            saveField(fieldToken.getText(), scopeParent.getType());
+        }
+    }
+
+
+    /*
+     * Check each non-final field definition from a class and fill nonFinalFields
+     *
+     * @param classDefinitionAST a class definition AST
+     */
+    private void fillClassFieldDefinitions(DetailAST classDefinitionAST) {
+        final DetailAST classObjBlockAst = classDefinitionAST.findFirstToken(TokenTypes.OBJBLOCK);
+
+        TokenUtil.forEachChild(classObjBlockAst, TokenTypes.VARIABLE_DEF, (definitionToken) -> {
+            final DetailAST variableModifiersAst = definitionToken.findFirstToken(TokenTypes.MODIFIERS);
+            if (!variableModifiersAst.branchContains(TokenTypes.FINAL)
+                && !Utils.hasIllegalCombination(variableModifiersAst)) {
+                nonFinalFields.add(definitionToken);
+            }
+        });
+    }
+
+    /*
+     * Get a node AST with parameters definition and return the list of all parameter names
+     * The set of parameters is created the first time an assignation is check within a method or constructor
+     * and we don't need to generate it again until visiting a different method or constructor.
+     * Field `currentScopeParameterSet` ensures we don't create the set multiple times for the same method/constructor
+     *
+     * @param parametersAST a TokenTypes.PARAMETERS
+     * @return a set of parameter names
+     */
+    private Set<String> getParameterSet (DetailAST parametersAST) {
```

Comment:
Are checkstyles run on this? I'm sure there shouldn't be a space after `Set` and before `(`